kernel/src/integration_main.rs

```diff
@@ -194,11 +194,14 @@ pub fn xmain() {
     // ... with 512 MiB of RAM per NUMA node ...
     for (nid, node) in MACHINE_TOPOLOGY.nodes().enumerate() {
         match nid {
-            0 => assert_eq!(node.memory().count(), 2),
-            _ => assert_eq!(node.memory().count(), 1),
+            0 => assert_eq!(node.memory().filter(|ma| !ma.is_non_volatile() & !ma.is_hotplug_region()).count(), 2),
```

Comment:
Why is the !is_hotplug_region necessary?

Reply:
We need an additional Qemu argument max-memory as explained in [PMEM in Qemu](https://docs.pmem.io/persistent-memory/getting-started-guide/creating-development-environments/virtualization/qemu#create-a-guest-vm). ACPI shows this as an additional hot-pluggable memory region that is a super-set of all the other memory regions. And while finding the number of usable memory regions, without this check, it will always be usable plus one.